File: EMS_BACKEND1/src/main/java/com/ems/service/EmployeeService.java
package com.ems.service;

import java.util.List;

import com.ems.dto.EmployeeDTO; // Import the new EmployeeDTO

public interface EmployeeService {

	EmployeeDTO createEmployee(EmployeeDTO employeeDTO); // Changed to use EmployeeDTO
	
	EmployeeDTO getEmployeeById(Long employeeId); // Changed return type to EmployeeDTO
	
	List<EmployeeDTO> getAllEmployees(); // Changed return type to List<EmployeeDTO>
	
	EmployeeDTO updateEmployee(Long employeeId, EmployeeDTO employeeDTO); // Changed to use EmployeeDTO
	
	void deleteEmployee(Long employeeId);
	
	EmployeeDTO getEmployeeByCompanyEmail(String companyEmail);
	
	//Employee getpersonalDetails(Long employeeId);

}

File: employee-management-frontend/src/services/EmployeeService.js
import axiosInstance from "../api/axiosConfig";

const EMPLOYEE_API_URL = "/employees";

const EmployeeService = {
  // ✅ Logged-in employee (NO ID)
  getEmployeeProfile: (employeeId) => {
    return axiosInstance.get(`${EMPLOYEE_API_URL}/profile/${employeeId}`);
  },

  updateEmployeeProfile: (employeeId, employeeData) => {
    return axiosInstance.put(`${EMPLOYEE_API_URL}/profile/${employeeId}`, employeeData);
  },

  // ✅ Admin APIs (ID required)
  getAllEmployees: () => {
    return axiosInstance.get(EMPLOYEE_API_URL);
  },

  getEmployeeById: (id) => {
    return axiosInstance.get(`${EMPLOYEE_API_URL}/${id}`);
  },

  createEmployee: (employeeData) => {
    return axiosInstance.post(EMPLOYEE_API_URL, employeeData);
  },

  updateEmployee: (id, employeeData) => {
    return axiosInstance.put(`${EMPLOYEE_API_URL}/${id}`, employeeData);
  },

  deleteEmployee: (id) => {
    return axiosInstance.delete(`${EMPLOYEE_API_URL}/${id}`);
  },
};

export default EmployeeService;
